torchinductor/lowering.py

```diff
@@ -1313,32 +1313,62 @@ def fn(idx):
     )
 
 
-@register_lowering(aten.scatter_add_, type_promote=False)
-def scatter_add_(self, dim: int, index, src):
+@register_lowering(aten.scatter_, type_promote=False)
+def scatter_(self, dim: int, index, src, **kwargs):
+    reduction_type = None
+    if kwargs:
+        if kwargs["reduce"] == "add":
+            reduction_type = "sum"
+        elif kwargs["reduce"] == "multiply":
+            reduction_type = "prod"
+        else:
+            assert False
+    return scatter_reduce_(self, dim, index, src, reduction_type)
+
+
+@register_lowering(aten.scatter_reduce_, type_promote=False)
+def scatter_reduce_(self, dim: int, index, src, reduction_type: str = None, **kwargs):
+    assert reduction_type is None or reduction_type in {"sum"}
     assert isinstance(self, TensorBox)
     assert "int" in str(index.get_dtype())
     assert 0 <= dim < len(self.get_size())
-    src = to_dtype(src, self.get_dtype())
+
     self.realize()
     V.graph.realize_users_of(self.get_name())
 
+    include_self = kwargs["include_self"] if kwargs else True
```

Comment:
you can potentially have `kwargs` that wouldn't include `include_self`, because other, non-kwarg-only args can be passed as kwargs. Better do `kwargs.get("include_self", True)`